### ISSUE TITLE
Can the font and font size follow the settings of the IDE itself?

### ISSUE THREAD

**Issue by RoseEnd** [carlrobertoh/CodeGPT#73]
Can the font and font size follow the settings of the IDE itself?

**Comment by carlrobertoh** [comment]
Hi, unfortunately the font doesn't change automatically once you change it from the IDE settings.
You need to either restart the IDE or create a new tab so the changes are reflected.
I will fix it in the upcoming days.

Thank you

**Comment by klaus7** [comment]
Love the plugin. But the font is always to big in my case. Would be awesome if this can be configured to be smaller. Isn't that big in any other IntelliJ tool window I use.

**Comment by pfrenssen** [comment]
The "Open in editor" function is a good workaround.

**Comment by klaus7** [comment]
I don't think so. The (main? one of its?) purpose of this whole plugin is to not be bothered with context switching, otherwise I could just copy and paste the contents with the prompts to ChatGPT directly. So when I have to use the  "Open in editor" function it adds another (very annoying also) context switch..

**Comment by carlrobertoh** [comment]
The "Open in editor" feature seems rather useless and I'm planning on removing it in future releases.

The font size of the tool window isn't relative to the IDE font sizes. Currently, I'm converting the IDE default font size into pixels and passing it on to the browser, which isn't an ideal solution.

That being said, I'm working on a POC to replace the JCEF browser with native Java Swing components. This should result in better UX and performance, including fixing the font sizes.

Sorry for the long delay.

**Comment by carlrobertoh** [comment]
@klaus7 Could you try upgrading the extension to the latest version and see if the problem still exists?

**Comment by klaus7** [comment]
@carlrobertoh 1.10.8, still the same issue: 
![image](https://github.com/carlrobertoh/CodeGPT/assets/5364438/46530c86-4bcc-4bca-a35d-5149b1798e96)

**Comment by carlrobertoh** [comment]
Latest version is 2.0.0

**Comment by klaus7** [comment]
Sorry, but I can't find it. 
![image](https://github.com/carlrobertoh/CodeGPT/assets/5364438/4231d1ca-0333-4d38-b826-18c568104082)

Also IntelliJ doesn't offer me a new plugin version.

**Comment by klaus7** [comment]
Found the 2.0.0 in the marketplace. Will report.

**Comment by klaus7** [comment]
@carlrobertoh The font size is smaller, but now it is impossible to read the code output, because it wierdly scrolls down and flickers. Hard to explain, but it looks like this:
![image](https://github.com/carlrobertoh/CodeGPT/assets/5364438/a816c574-e275-4e2d-a43c-d00b3dcc9e6e)

And I can't scroll up to see the code.

And it flickers:
[Screencast from 2023-06-06 14:26:47.webm](https://github.com/carlrobertoh/CodeGPT/assets/5364438/49d19ef1-bbb5-4cf2-ae28-81d9316ffe90)

**Comment by klaus7** [comment]
Sure:

Ubuntu 22.04.2 LTS

IntelliJ IDEA 2023.1.2 (Ultimate Edition)
Build #IU-231.9011.34, built on May 16, 2023
Runtime version: 17.0.6+10-b829.9 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.2.0-10007-tuxedo
Current Desktop: ubuntu:GNOME

I'm using the new UI of IntelliJ. I will be happy to provide more information if necessary. Thank you for your work!

**Comment by carlrobertoh** [comment]
Weirdly enough, I'm unable to reproduce it.

**Comment by klaus7** [comment]
@carlrobertoh Thank you, unfortunately that didn't change anything. I also upgraded IntelliJ, but that also didn't resolve the issue.

IntelliJ IDEA 2023.1.3 (Ultimate Edition)
Build #IU-231.9161.38, built on June 20, 2023

**Comment by carlrobertoh** [comment]
@klaus7 
Do you have "Show virtual space at the bottom of the file" turned ON by any chance? I was able to reproduce a similar issue with this setting turned on.
<img width="445" alt="Screenshot 2023-06-24 at 10 21 20" src="https://github.com/carlrobertoh/CodeGPT/assets/15891408/ee1b2437-0876-4ea6-a098-6743cd11cd29">

**Comment by klaus7** [comment]
@carlrobertoh Yes, I do. And yes, it works correctly when I turn off this setting. :) Nice finding!

**Comment by carlrobertoh** [comment]
@klaus The bug has been fixed in the [latest version](https://plugins.jetbrains.com/plugin/21056-codegpt/versions/stable/353051).

Furthermore, the font size can be changed from the IDE's settings: 
- Text - **Appearance & Behaviour | Appearance** 
- Code - **Editor | Font**
